app/controllers/application_controller.rb

```diff
@@ -1,10 +1,169 @@
 require './config/environment'
 
 class ApplicationController < Sinatra::Base
-
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
+  end
+
+get '/' do
+  if session[:user_id]
+     redirect '/tweets'
+   end
+  erb :index
+
+end
+
+get '/signup' do
+if session[:user_id]
+   redirect '/tweets'
+ end
+  erb :'users/create_user'
+end
+
+get '/login' do
+  if session[:user_id]
+     redirect '/tweets'
+   end
+  erb :'users/login'
+end
+
+
+
+post '/login' do
+  @user = User.find_by(:username => params[:username])
+  if @user && @user.authenticate(params[:password])
+  session[:user_id]= @user.id
+  redirect '/tweets'
+else
+  redirect '/login'
+end
+end
+
+
+get '/users/:id' do
+  @current_user = User.find_by(:username => params[:captures])
+  erb :'/users/show'
+end
+
+get '/tweets/new' do
+if  !is_logged_in?
+  redirect '/login'
+else
+  erb :'tweets/create_tweet'
+end
+end
+
+get '/tweets/:id' do
+  if  !is_logged_in?
+    redirect '/login'
+  else
+    @tweet = Tweet.find_by_id(params[:id])
+    erb :'/tweets/show_tweet'
+  end
+end
+
+get '/tweets/:id/edit' do
+  if  !is_logged_in?
+    redirect '/login'
+  else
+    @tweet = Tweet.find_by_id(params[:id])
```

Comment:
Shouldn't you verify a tweet belongs to the `current_user` (and not just that any user is logged in) before allowing them to edit?